app/models/invitation.rb

```diff
@@ -28,9 +36,30 @@ def to_param
     code
   end
 
+  def expired?
+    Time.now > expiration_date
+  end
+
+  def unexpired
+    raise RecordNotFound, "This invitation has already expired" if expired?
+    self
+  end
+
+  def self.generate_code
+    SecureRandom.urlsafe_base64 4
+  end
+
+  def notify!
+    Mumukit::Nuntius.notify_event! 'InvitationChanged', invitation: to_resource_h
+  end
+
   private
 
   def course_name
-    course.to_mumukit_slug.course
+    course.name
```

Comment:
Courses don't actually have a name, so I guess we should copy the previous logic to Course? as it is now, it breaks.

Reply:
You are right. Classroom calls it name, but domain calls it code. We should rename or alias it.